src/foam/nanos/blob/HttpBlobService.java

```diff
@@ -55,22 +55,19 @@ public void execute(X x) {
   }
 
   protected void download(HttpServletRequest req, HttpServletResponse resp) {
+    OutputStream os = null;
+
     try {
       String path = req.getRequestURI();
-      String id = path.substring(path.lastIndexOf("/") + 1);
+      String id = path.replaceFirst("/service/blobService/", "");
```

Comment:
Might someone ever want to mount this somewhere else? If so, you can find out where you're mounted by implementing NSpecAware so that you're provided your NSpec so that you can know where you're mounted. If not, then this is fine.

Reply:
Good point. Made the fix.